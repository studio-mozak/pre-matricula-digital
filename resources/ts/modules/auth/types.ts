export interface Auth {
  level: number;
  name: string;
  schools: string[];
}
