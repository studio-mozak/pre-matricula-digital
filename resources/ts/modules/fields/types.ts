import { Nullable } from '@/types';

export type SetValues = (fields: Record<string, unknown>) => void;

export interface FieldPageOption {
  id?: string;
  name: Nullable<string>;
  weight: number;
}

export interface FieldPage {
  id: Nullable<string>;
  name: Nullable<string>;
  description: Nullable<string>;
  group: Nullable<'RESPONSIBLE' | 'STUDENT'>;
  type: Nullable<string>;
  internal: boolean;
  options: FieldPageOption[];
  deleteOptions?: string[];
  required?: boolean;
}

export interface ErrorResponse {
  message: string;
}
