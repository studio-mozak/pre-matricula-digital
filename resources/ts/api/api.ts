import axios, { AxiosInstance } from 'axios';

const token = window.config ? window.config.token : '';

export const http: AxiosInstance = axios.create({
  baseURL: String(import.meta.env.VITE_APP_API_URL || ''),
  headers: {
    Authorization: `Token ${token}`,
  },
});

axios.interceptors.response.use(
  (res) => {
    /**
     * If the response from the server has the "errors" object or the object "data" is empty,
     * here it is forced to go to the catch block automatically.
     */
    if (
      (res.data.errors || Object.keys(res.data).length === 0) &&
      !(res.data instanceof Blob)
    ) {
      return Promise.reject({
        response: {
          data: res.data,
        },
      });
    }

    return res;
  },
  (err) => Promise.reject(err)
);

export const rest = <T>(
  type: 'get' | 'post' | 'put' | 'delete',
  url: string,
  params?: object
): Promise<T> => http[type](url, params);

export const graphql = <T>(data: object, config?: object): Promise<T> =>
  axios.post('/graphql', data, {
    headers: {
      Authorization: `Bearer ${token}`,
    },
    ...config,
  });
