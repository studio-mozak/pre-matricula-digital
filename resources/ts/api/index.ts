export * from './api';
export * as City from './services/graphql/city';
export * as Resource from './services/graphql/resource';
