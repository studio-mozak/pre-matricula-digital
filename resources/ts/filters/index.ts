import { Address, GoogleMapsMarker } from '@/types';
import {
  getFormattedDate,
  getFormattedDateTime,
  getFormattedTime,
} from '@/datetime';
import {
  joinAddress,
  joinAddressLine1,
  joinAddressLine2,
  joinAddressLine3,
  markerAddress,
} from '@/util';

export interface Filters {
  formatDateTime: (date: string) => string;
  formatDate: (date: string) => string;
  formatTime: (date: string) => string;
  joinAddress: (address: Address) => string;
  joinAddressLine1: (address: Address) => string;
  joinAddressLine2: (address: Address) => string;
  joinAddressLine3: (address: Address) => string;
  markerAddress: () => GoogleMapsMarker;
}

const filters: Filters = {
  formatDateTime(date: string) {
    return getFormattedDateTime(date);
  },
  formatDate(date: string) {
    return getFormattedDate(date);
  },
  formatTime(date: string) {
    return getFormattedTime(date);
  },
  joinAddress,
  joinAddressLine1,
  joinAddressLine2,
  joinAddressLine3,
  markerAddress,
};

export default filters;
