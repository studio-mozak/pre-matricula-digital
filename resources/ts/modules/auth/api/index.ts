/**
 * Exporting the Axios Instance
 */
export { http, graphql } from '@/api';

/**
 * Exporting all modules
 */
export * as AuthRest from '@/modules/auth/api/services/rest/auth';
