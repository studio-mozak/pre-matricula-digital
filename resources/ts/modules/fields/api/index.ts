/**
 * Exporting the Axios Instance
 */
export { http, graphql } from '@/api';

/**
 * Exporting all modules
 */
export * as Field from './services/graphql/field';
