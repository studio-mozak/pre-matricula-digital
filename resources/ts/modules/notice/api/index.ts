/**
 * Exporting the Axios Instance
 */
export { http, graphql } from '@/api';

/**
 * Exporting all modules
 */
export * as Notice from './services/graphql/notice';
