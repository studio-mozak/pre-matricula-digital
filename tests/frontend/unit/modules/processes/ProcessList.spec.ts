import { VueWrapper, flushPromises, mount } from '@vue/test-utils';
import { Process as ProcessApi } from '@/modules/processes/api';
import { ProcessGetList } from '@/modules/processes/types';
import ProcessList from '@/modules/processes/components/ProcessList.vue';
import VTooltip from 'v-tooltip';
import filters from '@/filters';
import { expect, vi } from 'vitest';

vi.mock('@/modules/processes/api');
vi.mock('vue-router/dist/vue-router.mjs', () => ({
  useRouter: () => ({
    push: spyPush,
  }),
}));

const spyPush = vi.fn();

const processMock: ProcessGetList = {
  processes: [
    {
      active: true,
      id: '1',
      key: '1',
      label: 'Educação infantil (Berçário a Nível II)',
      name: 'Educação infantil (Berçário a Nível II)',
      schoolYear: {
        year: 2021,
      },
      stages: [
        {
          id: '1',
          type: 'REGISTRATION_RENEWAL',
          name: 'Test',
          status: 'OPEN',
          startAt: '2020-10-14 00:00:00',
          endAt: '2022-10-19 23:59:00',
        },
      ],
    },
  ],
  years: [
    {
      key: 2021,
      label: '2021',
    },
  ],
  status: [
    {
      key: 'OPEN',
      label: 'EM ANDAMENTO',
    },
  ],
};

const onMounted = async (process: ProcessGetList, error = false) => {
  if (!error) {
    vi.mocked(ProcessApi.getList).mockResolvedValueOnce(process);
  } else {
    vi.mocked(ProcessApi.getList).mockRejectedValueOnce(process);
  }

  const wrapper = mount(ProcessList, {
    global: {
      provide: {
        $filters: filters,
      },
      stubs: ['router-link', 'x-card-section'],
      directives: {
        tooltip: VTooltip,
      },
    },
  });

  await wrapper.vm.$nextTick();

  await flushPromises();

  return wrapper;
};

describe('ProcessList', () => {
  let wrapper: VueWrapper<InstanceType<typeof ProcessList>>;

  beforeEach(async () => {
    wrapper = await onMounted(processMock);

    await wrapper.vm.$nextTick();
  });

  afterEach(() => {
    vi.clearAllMocks();
  });

  test('the component exists', () => {
    expect(wrapper).toBeTruthy();
  });

  test('should match snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  test('the initial variables must be loaded with the correct data', () => {
    expect(wrapper.vm.processes).toMatchObject(processMock.processes);
    expect(wrapper.vm.years).toMatchObject(processMock.years);
    expect(wrapper.vm.status).toMatchObject(processMock.status);
    expect(wrapper.vm.filteredProcesses).toMatchObject(processMock.processes);
    expect(wrapper.vm.filter).toMatchObject({
      year: null,
      status: null,
      id: null,
    });
  });

  test('should render 1 process on the screen', () => {
    const processesList = wrapper.findAll('[data-test="process-block"]');

    expect(processesList.length).toBe(1);
  });

  test('should render 1 stage on the screen', () => {
    const stagesList = wrapper.findAll('[data-test="stage-card"]');

    expect(stagesList.length).toBe(1);
  });

  test('when I click on "new process" button, should redirect to the new process page', async () => {
    const newProcessButton = wrapper.find('[data-test="new-process-button"]');

    newProcessButton.trigger('click');

    await flushPromises();

    expect(spyPush).toBeCalledTimes(1);
    expect(spyPush).toBeCalledWith({
      name: 'process.create',
      params: { newProcess: 'true' },
    });
  });

  test('should empty initial variables if occurs an error in the api', async () => {
    wrapper = await onMounted(
      {
        processes: [],
        years: [],
        status: [],
      },
      true
    );

    await wrapper.vm.$nextTick();

    expect(wrapper.vm.processes).toMatchObject([]);
    expect(wrapper.vm.years).toMatchObject([]);
    expect(wrapper.vm.status).toMatchObject([]);
    expect(wrapper.vm.filteredProcesses).toMatchObject([]);
    expect(wrapper.vm.filter).toMatchObject({
      year: null,
      status: null,
      id: null,
    });
  });
});
