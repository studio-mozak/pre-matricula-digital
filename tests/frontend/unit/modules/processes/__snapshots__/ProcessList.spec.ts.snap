// Vitest Snapshot v1

exports[`ProcessList > should match snapshot 1`] = `
<main>
  <div
    class="d-flex justify-content-between"
  >
    <h1>
      Processos
    </h1>
    <button
      class="x-btn x-btn-item non-selectable no-outline x-btn--standard x-btn--rectangle bg-primary text-white x-btn--actionable pmd-focusable pmd-hoverable x-btn--no-uppercase flex-row"
      data-test="new-process-button"
      role="button"
      tabindex="0"
      type="button"
    >
      <span
        class="x-btn__content text-center d-flex justify-content-center no-wrap text-no-wrap"
      >
        <span
          class="block"
        >
          Criar novo processo
        </span>
      </span>
      <transition-stub
        appear="false"
        css="true"
        name="x-transition--fade"
        persisted="false"
      >
        <!---->
      </transition-stub>
    </button>
  </div>
  <div
    class="mb-4 mt-4"
  >
    <div
      class="x-card x-card--flat no-shadow bg-primary"
    >
      <x-card-section-stub
        class="pl-2 pr-2"
        horizontal="false"
        tag="div"
      />
    </div>
  </div>
  <div
    class="row"
  >
    
    <div
      class="col-12 col-md-6 mt-4"
      data-test="process-block"
    >
      <div
        class="d-flex justify-content-between"
      >
        <router-link-stub
          class="text-truncate h4 text-blue-dark"
          to="[object Object]"
        />
        <!--v-if-->
      </div>
      
      <div
        class="x-card x-card--bordered x-card--hover mb-3"
        data-test="stage-card"
      >
        <x-card-section-stub
          class="pb-0"
          horizontal="false"
          tag="div"
        />
        <x-card-section-stub
          class="pt-0"
          horizontal="false"
          tag="div"
        />
      </div>
      
      <!--v-if-->
    </div>
    
    <!--v-if-->
  </div>
</main>
`;
